EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -686,47 +730,85 @@ void V2::World::importPotentialCountries()
 	dynamicCountries.clear();
 
 	std::ifstream v2CountriesInput;
-	v2CountriesInput.open("./blankMod/output/common/countries.txt");
-	if (!v2CountriesInput.is_open())
-		throw std::runtime_error("Could not open countries.txt. The converter may be corrupted, try downloading it again.");
+	std::vector<std::string> countriesFiles;
+	if (const auto& mod = theConfiguration.getVic2ModName(); !mod.empty())
+	{
+		if (commonItems::DoesFileExist(theConfiguration.getVic2ModPath() + "/" + mod  + "/common/countries.txt"))
+		{
+			countriesFiles.push_back(theConfiguration.getVic2ModPath() + "/" + mod  + "/common/countries.txt");
+		}
+		if (commonItems::DoesFileExist("configurables/" + mod  + "/common/countries.txt"))
+		{
+			countriesFiles.push_back("configurables/" + mod  + "/common/countries.txt");
+		}
+	}
+	countriesFiles.push_back("./blankMod/output/common/countries.txt");
 
-	auto dynamicSection = false;
-	while (!v2CountriesInput.eof())
+	for (const auto& countriesFile: countriesFiles)
 	{
-		std::string line;
-		getline(v2CountriesInput, line);
+		v2CountriesInput.open(countriesFile);
+		if (!v2CountriesInput.is_open())
+			throw std::runtime_error("Could not open " + countriesFile + ". The converter may be corrupted, try downloading it again.");
 
-		if (line[0] == '#' || line.size() < 3)
-			continue;
-		if (line.substr(0, 12) == "dynamic_tags")
+		auto dynamicSection = false;
+		while (!v2CountriesInput.eof())
 		{
-			dynamicSection = true;
-			continue;
+			std::string line;
+			getline(v2CountriesInput, line);
+
+			if (line[0] == '#' || line.size() < 3)
+				continue;
+			if (line.substr(0, 12) == "dynamic_tags")
+			{
+				dynamicSection = true;
+				continue;
+			}
+			importPotentialCountry(line, dynamicSection);
 		}
-		importPotentialCountry(line, dynamicSection);
+		v2CountriesInput.close();
 	}
-	v2CountriesInput.close();
 }
 
 void V2::World::importPotentialCountry(const std::string& line, bool dynamicCountry)
 {
 	auto tag = line.substr(0, 3);
 
-	auto newCountry = std::make_shared<Country>(line, dynamicCountry, partyNameMapper, partyTypeMapper);
-	potentialCountries.insert(std::make_pair(tag, newCountry));
-	if (dynamicCountry)
+	
+	if (countryMapper.getV2TagToModTagMap().find(tag) == countryMapper.getV2TagToModTagMap().end()
```

Comment:
rework this to use hpm field in country mappings, if HPM is enabled and field is present.